ideascube/serveradmin/catalog.py

```diff
@@ -611,8 +611,20 @@ def list_remotes(self):
         return sorted(self._remotes.values(), key=attrgetter('id'))
 
     def add_remote(self, id, name, url):
-        if id in self._remotes:
-            raise ValueError('There already is a "{}" remote'.format(id))
+        try:
+            existing_remote = self._remotes[id]
```

Comment:
If you want, but it will not save a lot of indent cause of the
```
if existing_remote:
    …
```
However, it is maybe better to use get(). try/catch may imply that is it the normal code path and, here, it is the integrity check/warning.